Tests/SourceKitLSPTests/BuildSystemTests.swift

```diff
@@ -155,14 +153,17 @@ final class BuildSystemTests: XCTestCase {
     let newSettings = FileBuildSettings(compilerArguments: args + ["-DFOO"])
     buildSystem.buildSettingsByFile[doc] = newSettings
 
-    let expectation = XCTestExpectation(description: "refresh")
-    let refreshedDiags = try await testClient.nextDiagnosticsNotification()
-    XCTAssertEqual(refreshedDiags.diagnostics.count, 0)
-    XCTAssertEqual(text, documentManager.latestSnapshot(doc)!.text)
-
     await buildSystem.delegate?.fileBuildSettingsChanged([doc])
 
-    try await fulfillmentOfOrThrow([expectation])
+    var receivedCorrectDiagnostic = false
+    for _ in 0..<Int(defaultTimeout) {
```

Comment:
Seems kind of weird to me to base this on timeout

Reply:
It makes sense to me. We wait 1 second for each notification and iterate this `defaultTimeout` times. That means that we wait for the correct diagnostic notification `defaultTimeout` seconds.

Reply:
Which only works if the number of notifications received is < defaultTimeout in seconds. Right now there's 2 so that's fine, but that's why it feels odd to me - the number of notifications we allow is dependent on the timeout value, rather than just receiving notifications up to a timeout/successful notification.

Reply:
Yes, the assumption is that number of notifications received ≪ default timeout. I think that’s a fair assumption to make. Happy to change it to `0..<30` though, like we do in other test case.